packages/mirinae/src/data-display/avatar/PAvatar.vue

```diff
@@ -0,0 +1,110 @@
+<script setup lang="ts">
+import { computed, reactive } from 'vue';
+
+import { get, isNull } from 'lodash';
+
+import type { SizeType, ColorType } from '@/data-display/avatar/type';
+import { AVATAR_SIZE, AVATAR_COLOR, ICON_SIZE } from '@/data-display/avatar/type';
+import PLazyImg from '@/feedbacks/loading/lazy-img/PLazyImg.vue';
+import PI from '@/foundation/icons/PI.vue';
+
+
+import colors from '@/styles/colors.cjs';
+
+interface AvatarProps {
+    color?: ColorType;
+    size?: SizeType;
+    icon?: string;
+    imgSrc?: string;
+}
+
+const props = withDefaults(defineProps<AvatarProps>(), {
+    imgSrc: '',
+    icon: 'ic_avatar-filled',
+    color: AVATAR_COLOR.INDIGO_200,
+    size: AVATAR_SIZE.MD,
+});
+
+const state = reactive({
+    imgSrc: computed(() => props.imgSrc),
+    iconName: computed(() => props.icon),
```

Comment:
nitpick: I think it's better to use these as `props` instead of making them `state`